File: problems/001-anagram-detection/rust/optimized.rs
use crate::json::JVal;

pub fn solve(c: &JVal) -> JVal {
    let s = c.get("s").as_str().as_bytes();
    let t = c.get("t").as_str().as_bytes();
    if s.len() != t.len() { return JVal::Bool(false); }
    let mut count = [0i32; 256];
    for &b in s { count[b as usize] += 1; }
    for &b in t {
        count[b as usize] -= 1;
        if count[b as usize] < 0 { return JVal::Bool(false); }
    }
    JVal::Bool(true)
}

File: tools/zig-behave-spike/b01/optimized.zig
const std = @import("std");
const json = std.json;

pub fn solve(alloc: std.mem.Allocator, c: json.Value) !json.Value {
    _ = alloc;
    const s = c.object.get("s").?.string;
    const t = c.object.get("t").?.string;
    if (s.len != t.len) return .{ .bool = false };
    var count = [_]i32{0} ** 256;
    for (s) |ch| count[ch] += 1;
    for (t) |ch| {
        count[ch] -= 1;
        if (count[ch] < 0) return .{ .bool = false };
    }
    return .{ .bool = true };
}
